ui/lib/pki/addon/controllers/tidy/index.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+import Ember from 'ember';
+import Controller from '@ember/controller';
+import { task, timeout } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+const POLL_INTERVAL_MS = 5000;
+
+export default class PkiTidyIndexController extends Controller {
+  @service store;
+  @service secretMountPath;
+
+  @tracked tidyStatus = null;
+
+  @task
+  *pollTidyStatus() {
```

Comment:
We should probably cancel when the user moves away from Tidy -- in the cluster route we do:
```
poll: task(function* () {
    while (true) {
      // do stuff
    }
  })
    .cancelOn('deactivate')
```
Not sure off the top of my head how that translates to class-based

Reply:
good catch thanks to claire for finding that `resetController` has a `isExiting` arg :)